src/timer.rs

```diff
@@ -21,48 +21,52 @@ pub use self::implementation::Timer;
 mod implementation {
     use action::Action;
     use itertools::Itertools;
-    use maidsafe_utilities::thread::{self, Joiner};
+    use maidsafe_utilities::thread;
     use std::collections::BTreeMap;
     use std::sync::{Arc, Condvar, Mutex};
     use std::time::{Duration, Instant};
     use types::RoutingActionSender;
 
     struct Detail {
+        next_token: u64,
         deadlines: BTreeMap<Instant, Vec<u64>>,
-        cancelled: bool,
+        use_count: u32,
     }
 
-    /// Simple timer.
+    /// A device for scheduling wake-ups, delivered via a message sender. This device can be
+    /// cloned with all wakeups sent via the same sender.
+    ///
+    /// The worker thread terminates when all associated `Timer` devices are destroyed.
     pub struct Timer {
-        next_token: u64,
         detail_and_cond_var: Arc<(Mutex<Detail>, Condvar)>,
-        _worker: Joiner,
     }
 
     impl Timer {
         /// Creates a new timer, passing a channel sender used to send `Timeout` events.
         pub fn new(sender: RoutingActionSender) -> Self {
             let detail = Detail {
+                next_token: 0,
                 deadlines: BTreeMap::new(),
-                cancelled: false,
+                use_count: 1,
             };
             let detail_and_cond_var = Arc::new((Mutex::new(detail), Condvar::new()));
             let detail_and_cond_var_clone = detail_and_cond_var.clone();
+
             let worker = thread::named("Timer", move || Self::run(sender, detail_and_cond_var));
-            Timer {
-                next_token: 0,
-                detail_and_cond_var: detail_and_cond_var_clone,
-                _worker: worker,
-            }
+            // TODO: confirm whether it is reasonable to disown the worker like this. Rationale is
+            // that it allows `Timer` to be cloned without a separate `Joiner`.
+            worker.detach();
```

Comment:
That is not the point -  it's just bad coding style. Just because it can be proved that a heap allocation of a raw ptr is being properly freed in all places doesn't mean one should use it (ownerless) instead of using a shared-ptr with ref count (shared owner) or a unique_ptr (Box) (one owner).

It's a maintainence hassle and bad coding in general. Also mentioned here: https://github.com/maidsafe/QA/blob/20110d496e1cfece1bfe80018f456aebf7ab511c/Documentation/Rust%20Style.md#threads